lib/pure/bitops.nim

```diff
@@ -68,55 +68,124 @@ when defined(nimHasalignOf):
 
   proc setMask*[T: SomeInteger](v: var T, mask: T) {.inline.} =
-    ## Returns ``v``, with all the ``1`` bits from ``mask`` set to 1
+    ## Returns ``v``, with all the ``1`` bits from ``mask`` set to 1.
+    runnableExamples:
+      var v = 0b0000_0011'u8
+      v.setMask(0b0000_1010'u8)
+      doAssert v == 0b0000_1011'u8
+
     v = v or mask
 
   proc clearMask*[T: SomeInteger](v: var T, mask: T) {.inline.} =
-    ## Returns ``v``, with all the ``1`` bits from ``mask`` set to 0
+    ## Returns ``v``, with all the ``1`` bits from ``mask`` set to 0.
+    runnableExamples:
+      var v = 0b0000_0011'u8
+      v.clearMask(0b0000_1010'u8)
+      doAssert v == 0b0000_0001'u8
+
     v = v and not mask
 
   proc flipMask*[T: SomeInteger](v: var T, mask: T) {.inline.} =
-    ## Returns ``v``, with all the ``1`` bits from ``mask`` flipped
+    ## Returns ``v``, with all the ``1`` bits from ``mask`` flipped.
+    runnableExamples:
+      var v = 0b0000_0011'u8
+      v.flipMask(0b0000_1010'u8)
+      doAssert v == 0b0000_1001'u8
+
     v = v xor mask
 
   proc setBit*[T: SomeInteger](v: var T, bit: BitsRange[T]) {.inline.} =
-    ## Returns ``v``, with the bit at position ``bit`` set to 1
+    ## Returns ``v``, with the bit at position ``bit`` set to 1.
+    runnableExamples:
+      var v = 0b0000_0011'u8
+      v.setBit(5'u8)
+      doAssert v == 0b0010_0011'u8
+
     v.setMask(1.T shl bit)
 
   proc clearBit*[T: SomeInteger](v: var T, bit: BitsRange[T]) {.inline.} =
-    ## Returns ``v``, with the bit at position ``bit`` set to 0
+    ## Returns ``v``, with the bit at position ``bit`` set to 0.
+    runnableExamples:
+      var v = 0b0000_0011'u8
+      v.clearBit(1'u8)
+      doAssert v == 0b0000_0001'u8
+
     v.clearMask(1.T shl bit)
 
   proc flipBit*[T: SomeInteger](v: var T, bit: BitsRange[T]) {.inline.} =
-    ## Returns ``v``, with the bit at position ``bit`` flipped
+    ## Returns ``v``, with the bit at position ``bit`` flipped.
+    runnableExamples:
+      var v = 0b0000_0011'u8
+      v.flipBit(1'u8)
+      doAssert v == 0b0000_0001'u8
+
+      v = 0b0000_0011'u8
+      v.flipBit(2'u8)
+      doAssert v == 0b0000_0111'u8
+
     v.flipMask(1.T shl bit)
 
   macro setBits*(v: typed, bits: varargs[typed]): untyped =
-    ## Returns ``v``, with the bits at positions ``bits`` set to 1
+    ## Returns ``v``, with the bits at positions ``bits`` set to 1.
+    ##
+    ## **Note:**
+    ## * ``v`` must be `var` variable.
+    runnableExamples:
+      var v = 0b0000_0011'u8
+      v.setBits(3, 5, 7)
+      doAssert v == 0b1010_1011'u8
+
     bits.expectKind(nnkBracket)
     result = newStmtList()
     for bit in bits:
       result.add newCall("setBit", v, bit)
 
   macro clearBits*(v: typed, bits: varargs[typed]): untyped =
-    ## Returns ``v``, with the bits at positions ``bits`` set to 0
+    ## Returns ``v``, with the bits at positions ``bits`` set to 0.
+    ##
+    ## **Note:**
+    ## * ``v`` must be `var` variable.
```

Comment:
instead, change signature to:
`macro clearBits*(v: var typed, bits: varargs[typed]): untyped =`
=> self documenting. ditto elsewhere

Reply:
Thanks. I changed.